### DISCUSSION TITLE
Combine transparency from clipping path (SVG) and transparency from alpha chanel

### DISCUSSION THREAD

**Discussion by kenobee** [ImageMagick/ImageMagick#4998] [Help] [answered]
PSD files could include a clippingpath (svg vector mask) and also a transparency (progressive grey level)
After hours I did not succeed to make PNG from PSD with both transparency
I try with IM6 and IM7 (Magick or convert)

convert /path/file.psd[0] -alpha transparent -clip -alpha opaque -geometry "1024x1024>" -density 72 /path/thumb.png

I try somthing like : -alpha set -background none -clip -fuzz 1% -transparent black -trim
To combine the alpha layer transparency and the clipping transparency but the result is bad with a black background

I send below :
the PSD file : clipalpha.pds
the bad PNG result : badresult.png
the wished PNG file : wishedresult.png

[clipalpha.psd.zip](https://github.com/ImageMagick/ImageMagick/files/8371782/clipalpha.psd.zip)

bad result
![badresult](https://user-images.githubusercontent.com/102373494/160621141-8f05a3d6-25cc-4e33-b62c-498ff313a3f5.png)

Result i wish
![wishpng](https://user-images.githubusercontent.com/102373494/160621169-5c97564e-4992-4aa8-96c6-6a7348226de9.png)

Screencopy of the result i wish
<img width="248" alt="viewtransparency" src="https://user-images.githubusercontent.com/102373494/160621089-6baafb16-dbfe-4f97-9b2e-36073a7e060c.png">

**Comment by snibgo** [comment]
When IM reads the image, some of its pixels are transparent. But then you use "-alpha transparent", which makes *all* pixels transparent.

This works for me:
```
magick clipalpha.psd[0] -alpha set -clip -alpha transparent +clip out.png
```
![out](https://user-images.githubusercontent.com/33812368/160713967-f678a621-a59c-4c2c-9530-0625a944834f.png)

**Reply by kenobee** [reply]
Hi thanks for your answer
unfortunately when i use the command line you suggest, i have not the same result !?
as well IM6 or IM7 
I use the OSX last version 7.1.0-29 Q16-HDRI arm 19841

i do not understand how you could have the result you send me with this command line ?

The IM7 documentation said 
1. -alpha set :
_Activates the alpha/matte channel. If it was previously turned off then it also resets the channel to opaque. **If the image already had the alpha channel turned on, it will have no effect.**_
as you said the alpha channel is already turn on, then no effect !

2. -clip :
_If a clipping path is present, it is applied to subsequent operations_
in this case you apply -alpha transparent onto the clip area !? 
this will erase the transparency from the blur rectangle in the middle of the bottle

conclusion the result i obtain with your command line is logical according to the documentation

I would like to combine (adding) the existing alpha channel with the clip (svg) channel
I try this, that work but very complicated :
magick alphaclip.psd[0] \( +clone -alpha set -background none -clip -fuzz 1% -transparent black -alpha extract +clip -alpha copy \) -compose copy_opacity -composite out.png

![thumb](https://user-images.githubusercontent.com/102373494/160771078-4821b3af-18e7-42c3-b30e-13e36fbf7a6e.png)

**Comment by fmw42** [comment]
I get the same bad result as the OP using snibgo's command on 7.1.0-29 Q16-HDRI on Mac OSX Sierra

However, the following works for me:

```
magick identify -quiet -format "%[8BIM:1999,2998:#1]" clipalpha.psd[0] |\
magick -quiet clipalpha.psd[0] +repage \
-profile /Users/fred/images/profiles/USWebCoatedSWOP.icc \
-profile /Users/fred/images/profiles/sRGB.icc \
\( -density 96 - -channel rgb -negate -blur 0x1 -level 50x100% +channel \) \
-alpha off -compose copy_opacity -composite \
-define tiff:alpha=associated \
out.png

```

![out](https://user-images.githubusercontent.com/13458218/160864085-083a1301-1203-44be-9fab-c279fa84c16d.png)

**Reply by kenobee** [reply]
Hi thank you for your time
if i download your final output the rectangle in the middle of the image is WHITE !
I would like a TRANSPARENT result

**Comment by fmw42** [comment, marked as answer]
I needed to extract the alpha channel, then the mask from the clip path, then combine them, then apply that to the base image.

```
magick identify -quiet -format "%[8BIM:1999,2998:#1]" clipalpha.psd[0] |\
magick -quiet clipalpha.psd[0] +repage \
\( -clone 0 -alpha extract +write mpr:alpha +delete \) \
\( -density 96 - -channel rgb -negate -blur 0x1 -level 50x100% +channel \) \
\( +clone mpr:alpha -compose multiply -composite \) \
-delete -2 \
-alpha off -compose copy_opacity -composite \
-define tiff:alpha=associated \
out.png

```

![out](https://user-images.githubusercontent.com/13458218/160886911-3c75278c-7f54-419f-9f2e-dbc2a5bc449b.png)

**Comment by kenobee** [comment]
Very good result, thanks
I found another way but i have a border with 1 pixel white arround the image, i must understand before using your command 
(Only one week i work with IM)
What do think about this command a little bit more simple ? :
magick -density 300 clipalpha.psd[0] \\( +clone -alpha set -background none -clip -transparent black -alpha extract +clip -alpha copy \\) -compose copy_opacity -composite -geometry "1024x1024>" -profile eciCMYK_v2.icc -profile eciRGB_v2.icc  out.png

![thumb](https://user-images.githubusercontent.com/102373494/160895315-dc7554fd-fbdd-4a16-9969-55e2944488b9.png)

**Reply by fmw42** [reply]
Add -fuzz 15% before -transparent:

`magick -density 300 clipalpha.psd[0] \( +clone -alpha set -background none -clip -fuzz 15% -transparent black -alpha extract +clip -alpha copy \) -compose copy_opacity -composite out.png
`
![out](https://user-images.githubusercontent.com/13458218/160902848-42f8ff3c-b945-43aa-8ba2-c331bfc122f9.png)

**Reply by kenobee** [reply]
very good result 
Thank you
